File: 0202-happy-number/0202-happy-number.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isHappy = function (n) {
  let seenMap = {};

  while (n !== 1 && !seenMap[n]) {
    seenMap[n] = true;
    n = squareSum(n);
  }
  return n === 1 ? true : false;
};

function squareSum(num) {
  let sum = 0;
  num = num.toString();
  for (let i = 0; i < num.length; i++) {
    sum += num[i] * num[i];
  }
  return sum;
}

File: 0202-happy-number/0202-happy-number.py
class Solution:
    
    def squareSum(self, num):
        sum = 0
        num = str(num)
        for i in range(len(num)):
            sum += int(num[i]) * int(num[i])
            
        return sum
    

    def isHappy(self, n: int) -> bool:
        seenSet = set()

        while (n != 1 and n not in seenSet):
            seenSet.add(n)
            n = self.squareSum(n)
        
        return n == 1
        

        
        